icc-x-api/icc-maintenance-task-x-api.ts

```diff
@@ -117,6 +119,18 @@ export class IccMaintenanceTaskXApi extends IccMaintenanceTaskApi {
     throw new Error('Cannot call a method that returns maintenance tasks without providing a user for de/encryption')
   }
 
+  deleteMaintenanceTask(maintenanceTaskIds: string): never {
+    throw new Error('Cannot call a method that returns maintenance tasks without providing a user for de/encryption')
+  }
+
+  deleteMaintenanceTaskWithUser(user: models.User, maintenanceTaskId: string): Promise<Array<DocIdentifier>> {
+    return this.crypto.getDataOwner(this.userApi.getDataOwnerOf(user)).then(({ dataOwner: hcp }) => {
+      return super
+        .getMaintenanceTask(maintenanceTaskId)
+        .then((mt) => (!!mt.delegations && hasAccessTo(hcp, mt.delegations) ? super.deleteMaintenanceTask(maintenanceTaskId) : []))
```

Comment:
Instead of returning an empty array, you can throw an Error saying the User does not have the right to delete the maintenance task